trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/CegarLoopFactory.java

```diff
@@ -221,16 +221,24 @@ private NwaCegarLoop<L> createFiniteAutomataCegarLoop(final IUltimateServiceProv
 					errorLocs, proofProducer, services, mTransitionClazz, stateFactoryForRefinement);
 		}
 
-		return switch (mPrefs.getFloydHoareAutomataReuse()) {
-		case EAGER -> new EagerReuseCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs, errorLocs,
-				proofProducer, services, Collections.emptyList(), rawFloydHoareAutomataFromFile, mTransitionClazz,
-				stateFactoryForRefinement);
-		case LAZY_IN_ORDER -> new LazyReuseCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs,
-				errorLocs, proofProducer, services, Collections.emptyList(), rawFloydHoareAutomataFromFile,
-				mTransitionClazz, stateFactoryForRefinement);
-		case NONE -> new NwaCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs, errorLocs,
-				proofProducer, services, mTransitionClazz, stateFactoryForRefinement);
-		};
+		switch (mPrefs.getFloydHoareAutomataReuse()) {
+		case EAGER:
+			return new EagerReuseCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs, errorLocs,
+					proofProducer, services, Collections.emptyList(), rawFloydHoareAutomataFromFile, mTransitionClazz,
+					stateFactoryForRefinement);
+		case LAZY_IN_ORDER:
+			return new LazyReuseCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs, errorLocs,
+					proofProducer, services, Collections.emptyList(), rawFloydHoareAutomataFromFile, mTransitionClazz,
+					stateFactoryForRefinement);
+		case NONE:
+			if (mPrefs.isParallelCegarLoop()) {
+				return new ParallelNwaCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs,
+						errorLocs, proofProducer, services, mTransitionClazz, stateFactoryForRefinement);
+			}
+			return new NwaCegarLoop<>(name, abstraction, root, csToolkit, predicateFactory, mPrefs, errorLocs,
+					proofProducer, services, mTransitionClazz, stateFactoryForRefinement);
+		}
+		throw new AssertionError("Unknown Setting: " + mPrefs.getFloydHoareAutomataReuse());
```

Comment:
As for other switches, please rewrite this such that it is an exhaustive switch again.